apps/examples/stack_protection/stack_protection_main.c

```diff
@@ -0,0 +1,119 @@
+/****************************************************************************
+ *
+ * Copyright 2020 Samsung Electronics All Rights Reserved.
+ * Author: Sangamanatha <sangam.swami@samsung.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+
+#include <sys/types.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <pthread.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+#define STACK_THREAD_SIZE			512
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+static pthread_addr_t stack_tc_thread()
+{
+	char ch;
+	int i = 0;
+	volatile char array[STACK_THREAD_SIZE];
+
+	while (1) {
+		printf("\nPress N - Test Stack access within range\n");
```

Comment:
There is no way for exit the test without overflow assertion.
How about adding new option for normal termination of test?

Reply:
yes added test case exit option